### DISCUSSION TITLE
Optional child components

### DISCUSSION THREAD

**Discussion by andreymosin** [arkivanov/Decompose#828] [Q&A]
Hello! I am doing a prototype of using Decompose as one of the possible options to migrate our native iOS/android apps to KMP.

If the structure of a component is known, all good, but there are a lot of situations in our app where we do need child components to be optional because at initialization we do not know final structure (API call returns it). Any tips how to approach it?

Example:
ScreenComponent -> ComponentA (fixed)
                                  -> ComponentB (optional)
                                  -> List of ComponentC (optional)
                                  -> List of any type components (optional)

The goal is to create / destroy components in runtime.

In iOS project we migrated to The Composable Architecture from Point-free and we absolutely love how we are able to split the big features into small ones and Decompose looks like a solution that would work in a similar fashion.

Thank you,
Andrey

**Comment by arkivanov** [comment]
For optional components you can try using the [Child Slot](https://arkivanov.github.io/Decompose/navigation/slot/overview/) navigation model. Let me know if it works for you.
